x/marker/keeper/marker.go

```diff
@@ -345,9 +345,10 @@ func (k Keeper) IncreaseSupply(ctx sdk.Context, marker types.MarkerAccountI, coi
 
 	inCirculation := sdk.NewCoin(marker.GetDenom(), k.bankKeeper.GetSupply(ctx, marker.GetDenom()).Amount)
 	total := inCirculation.Add(coin)
-	maxAllowed := k.GetParams(ctx).MaxTotalSupply
-	if total.Amount.Uint64() > maxAllowed {
-		return fmt.Errorf("requested supply %d exceeds maximum allowed value %d", total.Amount.Uint64(), maxAllowed)
+	maxAllowed := sdk.NewCoin(marker.GetDenom(), sdk.NewInt(int64(k.GetParams(ctx).MaxTotalSupply)))
```

Comment:
I think you want `sdk.NewIntFromUint64()` here.

Reply:
Nice! I did not see it as an available method.